### PR TITLE
[PB-534]: feature/Show the list of folders you have shared with other users

### PR DESCRIPTION
**Pull request by CandelR**
- Added functions to fetch private shared folders
- Added function to get public key with the email of an user
- Added function to get private shared folder data
- Added function to get list of users that has a specific shared folder shared

### CODE REVIEW

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/drive/share/index.ts [hunk 1]
```diff
@@ -145,6 +155,207 @@ export class Share {
...
+   * @param {UpdateRoleFolderPayload} options - The options for updating the user's role on the folder.
+   * @param {string} options.folderUUID - The unique identifier of the folder.
+   * @param {string} options.roleId - The identifier of the role to assign to the user.
+   * @param {string} options.userEmail - The email address of the user.
+   * @returns {Promise<UpdateRoleFolderResponse>} A promise that resolves when the user's role is updated.
+   */
+  public updateRoleFolder({
+    folderUUID,
+    roleId,
+    userEmail,
```
[the remark this answers is not in the document]
Comment (CandelR) [reply src/drive/share/index.ts:290]: Good catch! I had drive-server-wip on local out of date and I forgot to change it.

File: src/drive/share/types.ts [hunk 2]
```diff
@@ -86,3 +88,83 @@ export type ListShareLinksResponse = {
...
+
+export type PrivateSharedFolder = {
+  id: string;
+  folderId: string;
+  ownerId: string;
+  sharedWith: string;
+  encryptionKey: string;
+  createdAt: string;
+  updatedAt: string;
+  shared_with: string;
```
Comment (sg-gs) [inline src/drive/share/types.ts:144]: Same here with the camel case

**Review by sg-gs — APPROVED** [review]
Some of the paths are prone to change @CandelR

**Comment by CandelR** [comment]
> Some of the paths are prone to change @CandelR

What do you mean, that can change the paths of the enpdoints? Anyway, we will have to add the others as they are developed, we can change them later. We need this released for the staging environment.

**Review by sg-gs — APPROVED** [review]

**Pull request merged**